### PR TITLE
Less Boost, More Standard Library

### PR DESCRIPTION
**Pull request by brandonlehmann**
* Remove `boost::filesystem` in favor of `std::filesystem` in `FileMappedVector.h`

### CODE REVIEW

File: src/common/FileMappedVector.h [hunk 1]
```diff
@@ -940,17 +943,26 @@ namespace Common
...
+        /* Define the path to the backup file */
+        fs::path bakPath = m_path + ".bak";
+
+        /* Create a new temporary filename and path. Due to variations in
+           where different platforms create the temporary path to, we are
+           only going to use the filename of the unique path and prepend
+           the backup file's parent_path (aka. directory path) to come up
+           with the full temporary file path */
+        std::string uniqueName = std::tmpnam(nullptr);
+        fs::path uniquePath = uniqueName;
```
Comment (zpalmtree) [inline src/common/FileMappedVector.h]: Why are we converting to a `fs::path` here, then calling `.filename().string()` on it below - Couldn't we just do
```cpp
fs::path tmpPath = bakPath.parent_path().string() + uniqueName;
```

It doesn't look like we're using it elsewhere.

File: src/common/FileMappedVector.h [hunk 2]
```diff
@@ -940,17 +943,26 @@ namespace Common
...
+        fs::path tmpPath = bakPath.parent_path().string() + uniquePath.filename().string();
```
Comment (zpalmtree) [inline src/common/FileMappedVector.h]: Might we want to add a `-`, `_`, `.`, etc here? If our path is `foo` then the result file may be something like `foo.bakfileDjwifs` intead of `foo.bak_fileDjwifs`

File: src/common/FileMappedVector.h (hunk 1 above)
Comment (brandonlehmann) [reply src/common/FileMappedVector.h]: `tmpnam()` returns a `char*`, not a `path` and we just need the filename. Using just `uniqueName` results in a mess and possible write/permission errors not to mention it creates it in the environment temp directory.

File: src/common/FileMappedVector.h (hunk 2 above)
Comment (brandonlehmann) [reply src/common/FileMappedVector.h]: What? `foo.bak` would not be part of `tmpPath` due to the use of `parent_path()` which does not include the filename (`foo.bak`) thus the path would be `/some/path/fileDjwifs`

File: src/common/FileMappedVector.h (hunk 1 above)
Comment (zpalmtree) [reply src/common/FileMappedVector.h]: I'm not sure I follow. I understand it returns a `char *`, and then assigning to a `std::string` copies it. However, it doesn't appear to return a filepath to me, just a temporary name, going by the examples given. For this reason, I believe `uniquePath.filename().string()` is going to result in the same thing as `uniqueName`

File: src/common/FileMappedVector.h (hunk 2 above)
Comment (zpalmtree) [reply src/common/FileMappedVector.h]: Glanced over the `parent_path()`, ok.

File: src/common/FileMappedVector.h (hunk 1 above)
Comment (zpalmtree) [reply src/common/FileMappedVector.h]: Eh, nevermind, I read the docs wrong again.

Comment (brandonlehmann) [reply src/common/FileMappedVector.h]: > For this reason, I believe `uniquePath.filename().string()` is going to result in the same thing as `uniqueName`

It doesn't; that's the misconception. It returns an absolute path to the temporary file on *some platforms*

**Pull request merged**